### PR TITLE
Allow specifying user and host when adding keys

### LINKED ISSUE #2175: Specify user and/or host when adding key
Output of `restic version`
--------------------------
```
restic 0.9.4 compiled with go1.11.4 on linux/amd64
```

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------
It would be nice to be able to override the user and/or host values when adding a host to restic.

* Sometimes I am adding a key for a new host, but from a different host.
* Sometimes the short hostname on the new machine is ambiguous, so the key is added with the host `www`.  We might have multiple hosts with the short hostname `www`, so this is not useful.

What are you trying to do?
--------------------------
Add a new key with the FQDN of the current host, which does not seem possible.

### PR DESCRIPTION
**Pull request by pschultz**
This is an alternative to #2272, which proposes a special argument format to specify username and hostname, which I find unintuitive. This version uses normal flags (`--user` and `--host`) instead.

I copied the changelog entry directly from #2272, so credit goes to ahktenzero.

What is the purpose of this change? What does it change?
--------------------------------------------------------

Simplifies automated key managment by adding the `--user` and `--host` flags for the `key add` command.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

Prior PR in #2272, closes #2175.

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have added tests for all changes in this PR
- [x] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: internal/archiver/archiver_test.go
Comment (pschultz) [inline internal/archiver/archiver_test.go]: This test failed on my machine due to selinux:

```txt
--- FAIL: TestMetadataChanged (0.00s)
    testing.go:30: using low-security KDF parameters for test
    config.go:65: disabling check of the chunker polynomial
    archiver_test.go:2043: chdir to /tmp/restic-test-637943804
    archiver_test.go:2069: metadata does not match:
        {*restic.Node}.ExtendedAttributes:
                -: []restic.ExtendedAttribute(nil)
                +: []restic.ExtendedAttribute{{Name: "security.selinux", Value: []uint8{0x75, 0x6e, ..., 0x30, 0x00}}}
    panic.go:406: chdir back to /home/pschultz/src/restic/internal/archiver
FAIL
FAIL    github.com/restic/restic/internal/archiver      2.598s
```

I'm not sure why this test expected the ExtendedAttributes to be nil after snapshotting. [According to the manual ExtendedAttributes are preserved](https://github.com/restic/restic/blob/master/doc/manual_rest.rst#metadata-handling).

> Restic saves and restores most default attributes, including extended attributes like ACLs.

Comment (MichaelEischer) [reply internal/archiver/archiver_test.go]: My guess would be that it made the test work. If the node has no ExtendedAttribute, then the node create from NodeFromFileInfo has a value of `[]restic.ExtendedAttribute()`. The node loaded from the repository has `[]restic.ExtendedAttribute(nil)`. These two values differ, causing the test to fail.

The Lstat wrapper can't control the ExtendedAttributes, therefore your change seems to be the only easy way to fix the problem.

**Comment by ahktenzero** [comment]
I was checking on the PR I submitted for this feature and saw this one, I'm mostly in favour of merging this instead of mine but I'm not sure about the choice of option names. Restic uses `--host` in most commands for filtering by host, which is why I chose to use `key-hostname` instead and leave the possibility of having `--user` and `--host` options for filtering in `restic key list` or `restic key remove`.

Can I suggest `--new-user` and `--new-host` as the option names?  This would also be consistent with the existing `--new-password-file` option for the key commands.

**Comment by pschultz** [comment]
@ahktenzero, I have to disagree with you regarding the flag names.

Even if key filtering is added, it makes much more sense to me to use the same flag for filtering by some attribute as for setting that attribute. Having just `--user` and `--host` flags everywhere where there are user and hostnames involved is much easier to learn than a bunch of different flags that refer to the exact same thing. The `backup` and `snapshot list` commands don't use different flags for the hostname either, for instance.

The --new-password-file was necessary to avoid a collision with the global --password-file flag. When changing the password for a key one has to be able to provide both the existing and new password, so there have to be two flags. But that doesn't apply to user and hostnames.

File: doc/bash-completion.sh
Comment (fd0) [inline doc/bash-completion.sh]: This needs to be removed from the PR, the bash completion files are automatically generated for each release.

File: doc/man/restic-key.1
Comment (fd0) [inline doc/man/restic-key.1]: Same here, the manpage is automatically generated.

File: internal/archiver/archiver_test.go
Comment (fd0) [reply internal/archiver/archiver_test.go]: Leave it in for now, I don't see an easier way.

**Review by fd0 — CHANGES_REQUESTED** [review]
Awesome, thank you very much for your contribution! I'm very sorry that it took us so long to respond. I hope you're still interested.

Would you mind removing the changes to the man page and the completion file? And maybe rebase it to the current master branch?

Unfortunately I'm not allowed to push to the branch in your fork, otherwise I would've done that. Please let me know if you like any help. Thanks!

**Comment by rawtaz** [comment]
@pschultz You can give access for core contributors to push to your PR by editing the settings for that in the PR. See: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by pschultz** [comment]
I remove the changes to doc/* and rebased on master.

File: internal/archiver/archiver_test.go
Comment (MichaelEischer) [reply internal/archiver/archiver_test.go]: This problem has been fixed on master. Could you please rebase onto the current master and remove the test fix?

Comment (pschultz) [reply internal/archiver/archiver_test.go]: Done.

**Comment by samson4649** [comment]
@fd0 - is this waiting on anything?

**Comment by samson4649** [comment]
@fd0 - Does this need anything more before merging?

**Review by fd0 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
